### PR TITLE
Create Servo.app and dmg on mac package

### PR DESCRIPTION
**Pull request by cbrewster**
This creates Servo.app and puts it in a dmg along with a symlink to `/Applications`.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because creating packager

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11899)

### CODE REVIEW

File: python/servo/package_commands.py [hunk 1]
```diff
@@ -71,6 +82,71 @@ def package(self, release=False, dev=False, android=None, debug=False, debugger=
...
+                    install_name_tool(f, f.split('/')[-1], dir_to_app + '/Contents/MacOS/servo')
+
+            print("Writing run-servo")
+            # TODO: deduplicate this arg list from post_build_commands
+            servo_args = ['-w', '-b',
+                          '--pref', 'dom.mozbrowser.enabled',
+                          '--pref', 'dom.forcetouch.enabled',
+                          '--pref', 'shell.builtin-key-shortcuts.enabled=false',
+                          '--resources-path', '${0%/*}/../Resources',
+                          path.join('${0%/*}/../Resources', browserhtml_path.split('/')[-1], 'out', 'index.html')]
```
Comment (paulrouget) [inline python/servo/package_commands.py]: At this point, why not updating `prefs.json`?

Comment (ConnorGBrewster) [inline python/servo/package_commands.py]: Wouldn't this change the prefs for development runs aswell? or should we have a separate `prefs.json` just for packaging?

Comment (paulrouget) [inline python/servo/package_commands.py]: A separate json file maybe? In that json file, you could force webrender, the `-b` option, set all the other prefs and also add the location of the html page to load (`shell.homepage`).

Also, as for the resources path, we probably want to fix `resources_dir_path()` to return the `../Resources` directory.

With all of that, you wouldn't need a special script, but just run `./servo`.

I'm not sure what is best (a custom script or have a separate json file), but I think at least we should fix `resources_dir_path()`.

Comment (ConnorGBrewster) [inline python/servo/package_commands.py]: I fixed `resources_dir_path()` just by checking every directory backwards until it finds a resource folder. I just cannot figure out how to get the location of the bhtml index for `shell.homepage`. I have the index in `Servo.app/Contents/Resources/bhtml/out.index.html` but I cannot figure out what that path needs to be relative to.

Comment (ConnorGBrewster) [inline python/servo/package_commands.py]: We might have to keep the script just to get `shell.homepage` to work. Other than that, the prefs can be kept in a separate json.

**Comment by larsbergstrom** [comment]
```
./python/servo/package_commands.py:107: Line is longer than 120 characters

```

Otherwise, this lgtm.

**Comment by ConnorGBrewster** [comment]
After testing on some other people's macs it look like `install_name_tool` is not working. It still looks for the dylibs in the old location :/

**Pull request merged**